rskj-core/src/main/java/co/rsk/peg/RepositoryBtcBlockStoreWithCache.java

```diff
@@ -108,6 +147,31 @@ public synchronized void setChainHead(StoredBlock newChainHead) {
         if (cacheBlocks != null) {
             populateCache(newChainHead);
         }
+        setMainChainBlock(newChainHead.getHeight(), newChainHead.getHeader().getHash());
+    }
+
+    @Override
+    public Optional<StoredBlock> getInMainchain(int height) {
+        Optional<Sha256Hash> bestBlockHash = bridgeStorageProvider.getBtcBestBlockHashByHeight(height);
+        if (!bestBlockHash.isPresent()) {
+            logger.trace("[getInMainchain] Block at height {} not present in storage", height);
+            return Optional.empty();
+        }
+
+        StoredBlock block = get(bestBlockHash.get());
+        if (block == null) {
+            logger.trace("[getInMainchain] Block with hash {} not found in storage", bestBlockHash.get());
+            return Optional.empty();
+        }
+
+        logger.trace("[getInMainchain] Found block with hash {} at height {}", bestBlockHash.get(), height);
+        return Optional.of(block);
+    }
+
+    @Override
+    public void setMainChainBlock(int height, Sha256Hash blockHash) {
+        logger.trace("[setMainChainBlock] Set block with hash {} at height {}", blockHash, height);
+        bridgeStorageProvider.setBtcBestBlockHashByHeight(height, blockHash);
```

Comment:
There's a minimum height for btc block that I don't see neither validated nor enforced (I mean, preventing blocks outside that range from being saved here and making sure that starting at that height blocks will be saved)